steam/ext/dota2/state.py

```diff
@@ -0,0 +1,226 @@
+"""Licensed under The MIT License (MIT) - Copyright (c) 2020-present James H-B. See LICENSE"""
+
+from __future__ import annotations
+
+import asyncio
+from functools import partial
+from typing import TYPE_CHECKING, Any, NotRequired, TypedDict, Unpack
+
+from ..._gc import GCState as GCState_
+from ...app import DOTA2
+from ...enums import IntEnum
+from ...errors import WSException
+from ...id import _ID64_TO_ID32
+from ...state import parser
+from .models import PartialUser, User
+from .protobufs import client_messages, common, sdk, watch
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable, Sequence
+    from weakref import WeakValueDictionary
+
+    from ...protobufs import friends
+    from ...types.id import ID32, ID64, Intable
+    from .client import Client
+
+    # TODO: test this whole Kwargs thing, does it error if we don't provide some of them?
```

Comment:
Not required means the Client won't complain if they aren't passed to the initialiser. The defaults should be handled elsewhere